extension/intents/forms/formsContentScript.js

```diff
@@ -192,6 +193,28 @@ this.dictationContentScript = (function() {
     return focusDirection(-1);
   });
 
+  communicate.register("turnSelectionIntoLink", async message => {
+    const url = message.url;
+    let el;
+    if (!isPasteable(document.activeElement)) {
+      for (const selector of PASTE_SELECTORS) {
+        el = document.querySelector(selector);
+        if (el && isInViewport(el)) {
+          break;
+        }
+      }
+    } else {
+      el = document.activeElement;
+    }
+    if (el.hasAttribute("contenteditable")) {
+      replaceSelectedText(url);
+    } else if (el.hasAttribute("textarea")) {
+      const m = meta();
```

Comment:
This is running in the page where the link is being created, so it's going to pick up the title and URL of the current page. You want to replace it with the title and URL that you found for the query.